src/Spline.cpp

```diff
@@ -259,16 +259,21 @@ bool SplineBase::isNURBS() const
     return curve->ikind == 2 || curve->ikind == 4;
 }
 
-void SplineBase::interpolate(const vector< double >& points, 
-                             vector< double >& parametersOut, 
-                             const vector< double >& parametersIn, 
+void SplineBase::interpolate(const vector< double >& points,
+                             vector< double >& parametersOut,
+                             const vector< double >& parametersIn,
                              const vector< SplineBase::CoordinateType >& coord_types)
 {
     clear();
     start_param = 0.0;
     has_curvature_max = false;
 
-    int const point_count = points.size() / dimension;
+    unsigned int const point_count = points.size() / dimension;
+    if (!parametersIn.empty() && parametersIn.size() != point_count) {
+        throw std::invalid_argument(
```

Comment:
Hi @doudou ,
I am making a spline given position and velocity:
{base::geometry::SplineBase::CoordinateType::ORDINARY_POINT,  base::geometry::SplineBase::CoordinateType::DERIVATIVE_TO_PRIOR}
I have point_count = 2 * n, where n is the number of parameters. And I am getting this error.